PHPCompatibility/Sniffs/Classes/NewClassesSniff.php

```diff
@@ -1095,6 +1096,29 @@ class NewClassesSniff extends Sniff
         ],
     ];
 
+    /**
+     * Current file being scanned.
+     *
+     * @since 10.0.0
+     *
+     * @var string
+     */
+    private $currentFile;
```

Comment:
Just a nitpick, but otherwise the type would need to be `string|null`.

```suggestion
    private $currentFile = '';
```